### ISSUE TITLE
Changed from `select` to `computed` and my store no longer works

### ISSUE THREAD

**Issue by Awpteamoose** [ctrlplusb/easy-peasy#258]
I have just swapped `select` for `computed` to avoid using deprecated api and my store no longer works. I'm trying to pinpoint the issue, perhaps I'm using the library in a wrong way. Here's the error I'm getting:

```
VM27:59128 Uncaught (in promise) TypeError: Invalid property descriptor. Cannot both specify accessors and a value or writable attribute, #<Object>
    at Function.keys (<anonymous>)
    at ownKeys (eval at <anonymous> (index.js?ed34:90), <anonymous>:59128:57)
    at _objectSpread (eval at <anonymous> (index.js?ed34:90), <anonymous>:59128:411)
    at eval (eval at <anonymous> (index.js?ed34:90), <anonymous>:59133:447)
    at eval (eval at <anonymous> (index.js?ed34:90), <anonymous>:21050:3674)
    at simpleProduce (eval at <anonymous> (index.js?ed34:90), <anonymous>:21046:5150)
    at runActionReducerAtPath (eval at <anonymous> (index.js?ed34:90), <anonymous>:21050:3574)
    at reducerForActions (eval at <anonymous> (index.js?ed34:90), <anonymous>:21050:4586)
    at rootReducer (eval at <anonymous> (index.js?ed34:90), <anonymous>:21050:5401)
    at dispatch (eval at <anonymous> (index.js?ed34:90), <anonymous>:54025:406)
ownKeys @ VM27:59128
_objectSpread @ VM27:59128
(anonymous) @ base.js:48
(anonymous) @ create-component-store.js:20
simpleProduce @ create-store-internals.js:163
runActionReducerAtPath @ create-component-store.js:19
reducerForActions @ VM27:21050
rootReducer @ VM27:21050
dispatch @ redux.js:212
(anonymous) @ VM27:21050
(anonymous) @ VM27:21050
(anonymous) @ VM27:21050
(anonymous) @ index.js:11
actionCreator @ create-store-internals.js:293
_callee2$ @ index.jsx:48
tryCatch @ runtime.js:65
invoke @ runtime.js:303
prototype.(anonymous function) @ runtime.js:117
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
(anonymous) @ VM27:57767
(anonymous) @ VM27:57767
windowsSetup @ index.jsx:41
_callee3$ @ index.jsx:99
tryCatch @ runtime.js:65
invoke @ runtime.js:303
prototype.(anonymous function) @ runtime.js:117
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
Promise.then (async)
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
Promise.then (async)
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
Promise.then (async)
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
Promise.then (async)
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
Promise.then (async)
asyncGeneratorStep @ VM27:57767
_next @ VM27:57767
```

I've tracked it down to [this](https://github.com/ctrlplusb/easy-peasy/blob/v3/src/create-store-internals.js#L243) line.

Printing out `actionReducer`, `draft` and `action.payload` gives me this:
```javascript
ƒ (state,minimised){return _objectSpread({},state,{minimised:minimised});}
```

```javascript
Proxy {scope: ImmerScope, modified: false, finalized: false, assigned: {…}, parent: undefined, …}
[[Handler]]: Object
defineProperty: ƒ defineProperty()
deleteProperty: ƒ deleteProperty(state,prop)
get: ƒ get$1(state,prop)
getOwnPropertyDescriptor: ƒ getOwnPropertyDescriptor(state,prop)
getPrototypeOf: ƒ getPrototypeOf(target)
has: ƒ has(target,prop)
ownKeys: ƒ ownKeys(target)
set: ƒ set$1(state,prop,value)
setPrototypeOf: ƒ setPrototypeOf()
__proto__: Object
[[Target]]: Object
assigned: {}
base: {scale: 1, minimised: false, gameFocus: false, gameRunning: false, …}
copy: null
draft: Proxy {scope: ImmerScope, modified: false, finalized: false, assigned: {…}, parent: undefined, …}
drafts: {}
finalized: false
isManual: true
modified: false
parent: undefined
revoke: ƒ ()
scope: ImmerScope {drafts: Array(1), parent: null, canAutoFreeze: true, patches: null}
__proto__: Object
[[IsRevoked]]: false
```

```javascript
false
```

My model looks something like this:
```javascript
gameFocus: false
gameRunning: false
isRestored: ƒ ()
🙈computedSymbol🙈: {stateResolvers: Array(1)}
arguments: (...)
caller: (...)
length: 0
name: ""
prototype: {constructor: ƒ}
__proto__: ƒ ()
[[FunctionLocation]]: base.js:26
[[Scopes]]: Scopes[6]
localData: {version: "3.8.5", uuid: "edbae462-4f43-4bb8-845f-20c0386e94bc", notification: {…}, general: {…}, addon: {…}, …}
logMonitoring: false
logMonitoringHotkey: "Alt + Q"
logs: {entries: {…}, load: ƒ}
mainHotkey: "Alt + Q"
minimised: false
navigation: {page: "onboarding", logAnalysis: {…}, selectLog: ƒ, setMythicPlusSearch: ƒ, navigate: ƒ}
resources: {spells: {…}, icons: {…}, npcs: {…}, setCachedData: ƒ, startLoadingSpell: ƒ, …}
scale: 1
setGameFocus: ƒ (state,gameFocus)
setGameRunning: ƒ (state,gameRunning)
setLogMonitoring: ƒ (state,payload)
setLogMonitoringHotkey: ƒ (state,logMonitoringHotkey)
setMainHotkey: ƒ (state,mainHotkey)
setMinimised: ƒ (state,minimised)
setScale: ƒ (state,scale)
ui: {appNotification: {…}, notification: {…}, loginError: false, graph: {…}, addNotification: ƒ, …}
__proto__: Object
```

I'm building the model with object spread, also my actions return object spreads with store state. Something like this:
```javascript
setScale: action((state, scale) => ({ ...state, scale })),
setMinimised: action((state, minimised) => ({ ...state, minimised })),
setGameFocus: action((state, gameFocus) => ({ ...state, gameFocus })),
```

I'm using latest webpack & babel with nothing noteworthy in the config. It looks like the issue is related to immer somehow.

**Comment by Awpteamoose** [comment]
Replacing actions with mutating ones instead of "returning new state" ones seems to fix this, but it looks like a bug since the documentation says that returning new state is ok.

**Comment by Awpteamoose** [comment]
Here's a minimal repro:
```javascript
const easyPeasy = require("easy-peasy");

const model = {
	foo: 0,
	setFoo: easyPeasy.action((state, foo) => ({ ...state, foo })),
	doubleFoo: easyPeasy.computed((state) => state.foo * 2),
};

const store = easyPeasy.createStore(model);
store.dispatch.setFoo(5);
```

**Comment by ctrlplusb** [comment]
This is fixed in the new `3.0.0-rc.1`, but there is a breaking change requirement that you need to disable immer via the new store config `disableImmer` flag.

```javascript
import { createStore, action } from 'easy-peasy';

const model = {
  todos: [],
  addTodo: action((state, payload) => {
    // 👇 new immutable state returned
    return [...state, payload];
  })
}

const store = createStore(model, {
  disableImmer: true // 👈 set the flag
})
```
